geode-management/src/main/java/org/apache/geode/management/configuration/IndexType.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package org.apache.geode.management.configuration;
+
+/**
+ * Public ENUM to indicate type of Index on a Geode {@link Region} used by the Management API.
+ *
+ * @see Index
+ *
+ * @since Geode 1.12
+ */
+public enum IndexType {
+  FUNCTIONAL("RANGE"),
+  @Deprecated
+  HASH_DEPRECATED("HASH"),
+  PRIMARY_KEY("KEY");
+
+  private String name;
+
+  IndexType(String name) {
+    this.name = name;
+  }
+
+  public String getName() {
```

Comment:
`name` is an unfortunate name for this property, given that every enum already has a `name()` method. It's and confusing awkward that `name()` and `getName()` return different values.

Below, we call this `synonym`. Consider changing the field to `synonym` and this method to `getSynonym()`.

Reply:
if we don't define the synonyms inside IndexType class, then I agree with you. If the IndexType is aware of the synonyms, then it's the responsibility of the class itself to provide a translation.

Reply:
agree!

Reply:
@demery-pivotal I've applied this change.